packages/legend-graph/src/graph-manager/action/query/Query.ts

```diff
@@ -71,13 +73,41 @@ export class LightQuery {
   isCurrentUserQuery = false;
 }
 
+// use PartialQuery to update an existing query in DB with given attributes that are not null/undefined
+export class PartialQuery {
```

Comment:
Instead of creating a new class `PartialQuery`, could we just use the built-in TypeScript `Partial` utility type? i.e., `Partial<Query>`

Reply:
I did try to use PartialQuery, but I faced some issues like 
<img width="821" alt="Screenshot 2024-03-08 at 6 32 36 PM" src="https://github.com/finos/legend-studio/assets/73408381/f40b5c0f-dbef-4703-b52f-6e6ab6217a73">
😅
Haven't figured out a way to fix this

Reply:
Does that error still happen if you do this? Do we need to do `Query | Partial<Query>`?
```
export const V1_transformQuery = (
  metamodel: Partial<Query>,
): Partial<V1_Query> => {
```

Reply:
I added some if conditions to make it work 🥹😅